### PR TITLE
Generate book model and controller. Add validations to book model

### PR DESCRIPTION
**Pull request by TylerRockwell**
### Why?
https://github.com/smashingboxes/apprenticeship/tree/master/Curriculum/Backend/bookstore
Because what's a bookstore without books?

### What Changed?
Generated model and controller for books
Added validations to book model
Small configurations to test suite

### CODE REVIEW

File: app/models/book.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+class Book < ActiveRecord::Base
+  validates :title,           presence: true
```
Comment (iandonovan) [inline app/models/book.rb]: Part of me likes the verbosity here, but part of me likes the one-line solution, as well:

```ruby
validates_presence_of :title, :published_date, :author, :price
```

**Comment by iandonovan** [comment]
Looks good! I know you're trying to keep the PRs small, but I think you can make them _slightly_ larger in the future. :)

**Comment by TylerRockwell** [comment]
Thanks, @iandonovan . My first project was like 30+ changed files on every pr, so I'm trying to make sure I don't put people through that again. I'll continue working to find that happy middle.

File: app/models/book.rb (hunk 1 above)
Comment (TylerRockwell) [inline app/models/book.rb]: For some reason, I always forget that I can do this...

**Pull request merged**